app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Controller\Adminhtml\Stock;
+
+use Exception;
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\Api\DataObjectHelper;
+use Magento\Framework\Controller\Result\Redirect;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryApi\Api\Data\StockInterface;
+use Magento\InventoryApi\Api\Data\StockInterfaceFactory;
+use Magento\InventoryApi\Api\StockRepositoryInterface;
+
+/**
+ * Save Controller
+ */
+class Save extends Action
+{
+    /**
+     * @see _isAllowed()
+     */
+    const ADMIN_RESOURCE = 'Magento_Inventory::stock';
+
+    /**
+     * @var StockInterfaceFactory
+     */
+    private $stockFactory;
+
+    /**
+     * @var StockRepositoryInterface
+     */
+    private $stockRepository;
+
+    /**
+     * @var DataObjectHelper
+     */
+    private $dataObjectHelper;
+
+    /**
+     * @var StockSourceLinkProcessor
+     */
+    private $stockSourceLinkProcessor;
+
+    /**
+     * @param Context $context
+     * @param StockInterfaceFactory $stockFactory
+     * @param StockRepositoryInterface $stockRepository
+     * @param DataObjectHelper $dataObjectHelper
+     * @param StockSourceLinkProcessor $stockSourceLinkProcessor
+     */
+    public function __construct(
+        Context $context,
+        StockInterfaceFactory $stockFactory,
+        StockRepositoryInterface $stockRepository,
+        DataObjectHelper $dataObjectHelper,
+        StockSourceLinkProcessor $stockSourceLinkProcessor
+    ) {
+        parent::__construct($context);
+        $this->stockFactory = $stockFactory;
+        $this->stockRepository = $stockRepository;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->stockSourceLinkProcessor = $stockSourceLinkProcessor;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $requestData = $this->getRequest()->getParams();
+        if ($this->getRequest()->isPost() && !empty($requestData['general'])) {
+            try {
+                $stockId = !empty($requestData['general'][StockInterface::STOCK_ID])
+                    ? $requestData['general'][StockInterface::STOCK_ID] : null;
```

Comment:
Not quite, you [might get a notice](https://stackoverflow.com/questions/6052763/php-undefined-index-using-ternary-operator).
However, as of PHP 7, it's possible to use the [`??` (null coalescing) operator](http://php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce). Assuming Magento2 will drop support of PHP 5.6 soon, it might become a feasible alternative soon.